scalafix/rules/src/main/scala/rsc/rules/semantics/Symtabs.scala

```diff
@@ -2,6 +2,34 @@
 // Licensed under the Apache License, Version 2.0 (see LICENSE.md).
 package rsc.rules.semantics
 
+import scala.meta.internal.{semanticdb => s}
+import scala.meta.internal.semanticdb.Scala.{Names => n}
+
 trait Symtabs {
   type Symtab = scala.meta.internal.symtab.SymbolTable
+
+  implicit class SymtabOps(symtab: Symtab) {
+
+    def equivalent(env: Env, name: Option[n.Name], sym: String): Boolean = {
+      name
+        .flatMap(x => symtab.info(env.lookup(x)))
+        .exists { info =>
+          info match {
+            case s.SymbolInformation(
+                _,
+                _,
+                _,
+                _,
+                _,
+                s.TypeSignature(_, s.TypeRef(_, loSym, _), s.TypeRef(_, hiSym, _)),
+                _,
+                _) =>
```

Comment:
I'd do `info.signature match { ... }`. That would avoid the need for a humongous case clause in the current pattern match.